word-search.py

```diff
@@ -257,7 +257,27 @@ def word_search():
                 print("Invalid word. Try again.")
         else:
             print("Invalid word. Try again.")
-    print(f"Your final score is: {currentScore}.")  
-  
 
-word_search()
\ No newline at end of file
+        if timeroption == 4:  # If untimed game is selected, ask the player if they want to continue or restart
+            continue_game = continue_or_restart()  # Ask the player whether to continue or restart after each guess
+            if not continue_game:
+                print("Thank you for playing!")
+                break  # Exit the game if the player chooses not to continue
+        else:
```

Comment:
I think it should be play_again == 'y' to allow the game to "restart". Also, ensure to handle the 'n' input explicitly so that the player can opt not to continue. For unexpected inputs like 'w', prompt the user again to provide valid input ('y' or 'n') to maintain clarity and avoid unintended behavior.